test/fast/Unit tests/nvm_ensure_posix_path

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+
+cleanup () {
+  unset -f die cleanup
+  unset NVM_SOURCE out
+  # sudo rmdir '/mnt/z' 2>/dev/null
+  # sudo rmdir '/mnt' 2>/dev/null
+  # sudo rmdir '/drives/z' 2>/dev/null
+  # sudo rmdir '/drives/w' 2>/dev/null
+  # sudo rmdir '/drives' 2>/dev/null
+}
+die () { echo "$@" ; cleanup ; exit 1; }
+
+\. ../../../nvm.sh
+
+set -ex
+
+# sudo mkdir -p '/mnt/z'
+# sudo mkdir -p '/drives/z'
+# sudo mkdir -p '/drives/w'
+
+# nvm_ensure_posix_path is available
+type nvm_ensure_posix_path > /dev/null 2>&1 || die 'nvm_ensure_posix_path is not available'
+
+# nvm_ensure_posix_path with no parameter fails
+nvm_ensure_posix_path 2>&1 | grep "Need the location as first parameter" >/dev/null || die 'nvm_ensure_posix_path without parameter fails'
+
+# nvm_ensure_posix_path does not change a path wich is already posix
+[ "$(nvm_ensure_posix_path '/')" = "/" ] || die 'nvm_ensure_posix_path changed "/"'
```

Comment:
```suggestion
[ "$(nvm_ensure_posix_path '/')" = '/' ] || die 'nvm_ensure_posix_path changed "/"'
```
and throughout; let’s avoid double quotes when not interpolating

Reply:
indeed